### PR TITLE
Ensure that bucket picker respect RAM-limit for buckets

### PR DESCRIPTION
**Pull request by philippotto**
If the GPU is able to allocate more than 5000 buckets for a layer, the bucket picker code could theoretically cause that more than 5000 buckets were fetched by the pullqueue. However, each `DataCube` has a hard limit of 5000 which meant that each new bucket kicked another bucket out of the `DataCube`. This PR makes the bucket picker respect the hard limit of 5000 and also reports a proper error to airbrake.

I temporarily reduced the hard limit from 5000 to 100 and checked that the change really improves things. Indeed, before the fix, the rendered output was very spotty (lots of randomly distributed black buckets). After the fix, this was way better.
However, I still noticed some "forced bucket collection" messages. From my understanding, the prefetching is at fault here, since it doesn't respect any bucket limits at all. I think, there is no easy fix for this. Instead, we probably should refactor the entire logic around which buckets are pulled / kept in RAM / kept on GPU so that all limits can be respected correctly.

For now, this fix should help a bit (hopefully) and we can observe airbrake to see whether the other aspect is really happening (or if it's only a theoretical problem (or if it's the main problem)).

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- see explanation above

### Issues:
- contributes to #5000

------
- [ ] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Updated [(unreleased) migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes
- [ ] Needs datastore update after deployment
- [X] Ready for review

### CODE REVIEW

File: frontend/javascripts/oxalis/model/bucket_data_handling/data_cube.js [hunk 1]
```diff
@@ -259,17 +260,22 @@ class DataCube {
 
   addBucketToGarbageCollection(bucket: DataBucket): void {
     if (this.bucketCount >= this.MAXIMUM_BUCKET_COUNT) {
-      for (let i = 0; i < 2 * this.bucketCount; i++) {
```
Comment (philippotto) [inline frontend/javascripts/oxalis/model/bucket_data_handling/data_cube.js:262]: I think, earlier versions of `shouldCollect()` had side effects which is why this code iterated twice over all buckets. However, now, the method is sideeffect-free which is why a single iteration should be enough.

File: frontend/javascripts/oxalis/constants.js [hunk 2]
```diff
@@ -178,6 +178,9 @@ const Constants = {
 
   MIN_TREE_ID: 1,
   MIN_NODE_ID: 1,
+
+  // Maximum of how many buckets will be held in RAM (per layer)
+  MAXIMUM_BUCKET_COUNT_PER_LAYER: 100,
```
Comment (philippotto) [inline frontend/javascripts/oxalis/constants.js:183]: todo: revert back to 5000

File: .circleci/not-on-master.sh [hunk 3]
```diff
@@ -1,8 +1,8 @@
 #!/usr/bin/env bash
 set -Eeuo pipefail
 
-if [ "${CIRCLE_BRANCH}" == "master" ]; then
+# if [ "${CIRCLE_BRANCH}" == "master" ]; then
```
Comment (daniel-wer) [inline .circleci/not-on-master.sh:4]: Reminder to revert this as well.

**Review by daniel-wer — COMMENTED** [review]
Thanks for looking into this, LGTM!

> However, I still noticed some "forced bucket collection" messages. From my understanding, the prefetching is at fault here, since it doesn't respect any bucket limits at all. I think, there is no easy fix for this. Instead, we probably should refactor the entire logic around which buckets are pulled / kept in RAM / kept on GPU so that all limits can be respected correctly.

Agreed, the additional error logging in this PR should help us decide on the priority of that :) Another possibility would be to factor in a certain percentage of prefetched buckets in the calculation in `getBucketCapacity`. For example, the method could return only 0.75 times the theoretical capacity to allow space for prefetched buckets. I'm not sure whether such a static percentage exists and is robust, though.

**Review by daniel-wer — APPROVED** [review]
:+1:

**Comment by philippotto** [comment]
> Another possibility would be to factor in a certain percentage of prefetched buckets in the calculation in getBucketCapacity. For example, the method could return only 0.75 times the theoretical capacity to allow space for prefetched buckets. I'm not sure whether such a static percentage exists and is robust, though.

Yes, that could be an easy workaround :+1:

**Pull request merged**